### PR TITLE
feat(aws): Adding support for session stickiness for clbs upsert op

### PR DESCRIPTION
**Pull request by jeyrschabu**
- Added ability to define session stickiness for listeners

### CODE REVIEW

**Comment by jeyrschabu** [comment]
Currently, deck doesn't allow adding session stickiness of classic load balancer listeners. You have to do that in the console. Of course if you edit listeners, clouddriver gladly recreate listeners and all configs related to that listener are lost. This could lead to bad surprises if your application does indeed rely on session stickiness for requests.

This PR exposes the ability to add session stickiness in the request to upsert a classic load balancer.

**Comment by jeyrschabu** [comment]
Sample request:
```
"listeners" : [
		{
			"internalProtocol" : "HTTP",
			"sslCertificateId" : "arn:to:cert",
			"externalPort" : 443,
			"externalProtocol" : "HTTPS",
			"internalPort" : 80,
			"sslCertificateName" : "cert-test-blah",
			"sslCertificateType" : "iam"
		},
		{
			"externalPort" : 80,
			"internalPort" : 7001,
			"externalProtocol" : "HTTP",
			"internalProtocol" : "HTTP",
			"cookiePolicies": [
				{
					"policyName": "cookieStickinessTest",
					"cookieName": "yummyCookie",
					"type": "APP",
					"durationSeconds": 10
				}
			]
			
		}
	]
```

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertAmazonLoadBalancerClassicDescription.java [hunk 1]
```diff
@@ -176,5 +178,26 @@ public String getSslCertificateId() {
...
+    public static class CookiePolicy {
+      public enum CookieType {
+        APP,
+        LB
+      }
+
+      public CookieType cookieType;
+      public String policyName;
+      public String cookieName;
+      public long durationSeconds;
```
Comment (jeyrschabu) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertAmazonLoadBalancerClassicDescription.java]: `durationSeconds` available for `LBCookieStickinessPolicy` or  `CookieType.LB`

Comment (robzienert) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertAmazonLoadBalancerClassicDescription.java]: This PR comment should definitely be a comment on the property itself.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertAmazonLoadBalancerClassicDescription.java [hunk 2]
```diff
@@ -176,5 +178,26 @@ public String getSslCertificateId() {
...
+
+    public List<CookiePolicy> getCookiePolicies() {
+      return cookiePolicies;
+    }
+
+    public void setCookiePolicies(List<CookiePolicy> cookiePolicies) {
+      this.cookiePolicies = cookiePolicies;
+    }
+
+    @Data
```
Comment (robzienert) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertAmazonLoadBalancerClassicDescription.java]: `@Data` annotation, but all of the properties are defined as `public`? What's the point?

Comment (jeyrschabu) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertAmazonLoadBalancerClassicDescription.java]: re: no test - Oh yeah there will be tests, wanted to see if this was good direction from folks.

**Comment by robzienert** [comment]
Ping.

**Comment by jeyrschabu** [comment]
Picking this back up.

**Comment by jeyrschabu** [comment]
Simplified to just apply the policy names that are passed in the request since this is only for CLBs.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy [hunk 3]
```diff
@@ -258,8 +269,32 @@ class UpsertAmazonLoadBalancerAtomicOperation implements AtomicOperation<UpsertA
     operationResult
   }
 
+  /**
+   * Ensures policies set in the request are applied to the load balancer
+   */
+  private static void ensureSetLoadBalancerListenerPolicies(
+    UpsertAmazonLoadBalancerClassicDescription loadBalancer, AmazonElasticLoadBalancing loadBalancing) {
+    for (UpsertAmazonLoadBalancerClassicDescription.Listener listener :  loadBalancer.listeners) {
+      final List<String> policyNames = listener.policyNames ?: []
```
Comment (alanmquach) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy]: This works for me!

**Comment by jeyrschabu** [comment]
Waiting for a thumbs up to test this with https://github.com/spinnaker/deck/pull/7897#pullrequestreview-359154274

**Pull request merged**